src/uct/ib/mlx5/dv/ib_mlx5dv_md.c

```diff
@@ -46,6 +47,54 @@ uct_ib_mlx5_md_check_odp_common(uct_ib_mlx5_md_t *md, const char **reason_ptr)
     return 1;
 }
 
+static void uct_ib_mlx5dv_check_direct_nic(struct ibv_context *ctx,
+                                           uct_ib_device_t *dev,
+                                           uct_ib_mlx5_md_t *md,
+                                           const uct_ib_md_config_t *md_config)
+{
+#if HAVE_DIRECT_NIC
+    char sys_path[PATH_MAX];
+    char dev_name[64];
+    int ret;
+    ucs_status_t status;
+
+    if (!md_config->ext.direct_nic) {
+        goto out;
+    }
+
+    ret = mlx5dv_get_data_direct_sysfs_path(ctx, sys_path, sizeof(sys_path));
+    if (ret != 0) {
+        ucs_debug("%s: mlx5dv_get_data_direct_sysfs_path() failed: ret=%d",
+                  uct_ib_device_name(&md->super.dev), ret);
+        goto out;
+    }
+
+    /* Create a DMA specific device from topology perspective */
+    snprintf(dev_name, sizeof(dev_name), "%s_direct",
+             uct_ib_device_name(&md->super.dev));
+    md->direct_nic_sys_dev = ucs_topo_get_sysfs_dev(dev_name, sys_path, 0);
+
+    status = ucs_topo_sys_device_set_sys_dev_aux(dev->sys_dev,
+                                                 md->direct_nic_sys_dev);
+    if (status != UCS_OK) {
+        ucs_debug("ucs_topo_sys_device_set_sys_dev_aux failed: %s",
+                  ucs_status_string(status));
+        goto out;
+    }
+
+    ucs_debug("%s: Direct NIC is supported sys_path='%s%s' "
+              "sys_dev=%u sys_dev_aux=%u",
+              uct_ib_device_name(&md->super.dev),
+              (sys_path[0] != 0) ? "/sys" : "", sys_path, dev->sys_dev,
+              md->direct_nic_sys_dev);
+    return;
+out:
+#endif
+    md->direct_nic_sys_dev = UCS_SYS_DEVICE_ID_UNKNOWN;
+    ucs_debug("%s: Direct NIC support disabled",
```

Comment:
it should be an error if direct_nic eplicitly requested. Can address in another PR